### PR TITLE
feat: options to display continuous sliders

### PR DESCRIPTION
**Pull request by Alan-Kuan**
Hello.

I like what this plugin does, but what annoyed me was that the appearance of the sliders looked uncontinuous because the background color of the points were empty.
Thus, I added some options to make it possible to display continuous sliders.
In order not to introduce breaking changes to everyone's configs, the default behavior is same as before.

The options I added:
- empty_point_bg
  - default: `true`
  - whether to give points background colors
- point_color_on_dark
  - default: `"#ffffff"`
  - point color on a dark background
  - only works if empty_point_bg is false
- point_color_on_light
  - default: `"#000000"`
  - point color on a light background
  - only works if empty_point_bg is false

The followings are some screenshots.

20230603_22h52m44s_grim

20230603_22h53m14s_grim

Besides, I updated the document, so you can also take a look.

Please tell me if there's anything I missed.

### CODE REVIEW

File: lua/ccc/ui.lua [hunk 1]
```diff
@@ -322,15 +326,25 @@ function UI:highlight()
...
-        hex = point_color
+      local new_value = (j - 0.5) / bar_len * (max - min) + min
+      local hex = self.color:hex(i, new_value)
+      local hi = {}
+      if j == point_idx then
+        hi.fg = hex
+        if not empty_point_bg then
+          local RGB = self.color:get_rgb()
+          local _, _, L = unpack(convert.rgb2hsl(RGB))
+          hi.fg = L < 0.6 and point_color_on_dark or point_color_on_light
```
Comment (uga-rosa) [inline lua/ccc/ui.lua:337]: You can use this to calculate brightness. Right now it's a local function, but you can make it public if you want. You would need to modify it a bit, though, as you want to calculate from RGB values, not from strings.
https://github.com/uga-rosa/ccc.nvim/blob/78aed26e18a087501be0475443b5a623adbf6290/lua/ccc/utils/init.lua#L117-L126

**Comment by uga-rosa** [comment]
As for what you want to do, I understand and I think the policy is good.

File: lua/ccc/ui.lua (hunk 1 above)
Comment (Alan-Kuan) [reply lua/ccc/ui.lua:337]: I see. Let me give it a try.

**Comment by Alan-Kuan** [comment]
@uga-rosa I finished. Please take a look.

**Comment by uga-rosa** [comment]
LGTM
Thanks for your work and contribution!

**Pull request merged**